### PR TITLE
Convert project to poetry

### PR DESCRIPTION
**Pull request by mweinelt**
Poetry offers a more convincing story on dependency and enrivonment management, as well as a an integrated build and publish workflow.

The `pyproject.toml` file is the latest configuration for python project tooling and the way forward.

I have no idea how to actually test the GitHub action.

### CODE REVIEW

**Comment by mweinelt** [comment]
Quick start with poetry:
- Install dependencies: `poetry install [--no-dev]`
- Add a new dependency: `poetry add <package>`
- Update all dependencies: `poetry update`
- Build the project: `poetry build`
- Publish the project: `poetry publish`
- Run something in poetrys environment: `poetry run <cmd>`,
  - e.g. `poetry run wgkex-broker`, as defined in `pyproject.toml`

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[tool.poetry]
+name = "wgkex"
+version = "0.1.0"
+description = "Wireguard Connection Broker"
+repository = "https://github.com/freifunkMUC/wgkex"
+readme = "README.md"
+license = "GPL-2.0-only"
```
[the remark this answers is not in the document]
Comment (awlx) [reply pyproject.toml:7]: Sounds good.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+authors = []
```
[the remark this answers is not in the document]
Comment (awlx) [reply pyproject.toml:8]: That makes sense :).

**Comment by awlx** [comment]
Makes sense for me I vote for merge :).

**Comment by mweinelt** [comment]
> I have no idea how to actually test the GitHub action.

Guess we cross that bridge, when we come to it.

**Pull request merged**